### PR TITLE
Rk/rethink scan page

### PR DESCRIPTION
**Pull request by rkaufman13**
For all's consideration:
This significantly refactors the scan order page, moving the ability to edit into the scan page and making the scan page more accessible from the homepage.
I'm leaving this as a draft because the one thing I can't figure out is how to refresh the offices dropdown after the state changes. You'll see in selectField there's a conditional that fires if usa_state is the select that's been edited, and if you console.log the value of `optionDistricts` after a change, the districts are correct--I just can't get the component to rerender. Makes me think a `useEffect` is in order but I've always been a bit rusty on those.
Folks are welcome to take this and run with it, or leave suggestions.

### CODE REVIEW

**Comment by mertbagt** [comment]
So, the entirety of this comment is just my opinion, not objective fact, and should be taken with a grain of salt.

Overall, I think this is a very ambitious PR that does a lot of cool things, but I feel like it's trying to do a lot all at once and could maybe have been a 3 part PR series based on specific, more easily digestible code concerns.

### subPR 1: Refactoring Edit, Scan, Print buttons

This is great; I love this and think it's *extremely brilliant*.  I would have wired the Edit and Print buttons to their respective components and submitted this as stage 1 (even though I realize you intend to refactor those respective components) because it

- submits something that both works and is an immediate and wonderful upgrade
- it doesn't get compounded with the complexity of working through all the other stuff *all at once*
- the edit/print buttons can be easily redirected as needed as those other code concerns get addressed

### subPR 2: upgrading handwritten forms to Formik

I'm not personally familiar with Formik but I'm sure it has a ton of potential to handle forms, user error messaging, etc.

The components of the app that use forms are: AddOrder, EditOrder, Profile, AddUser, Login/LoginSubComponent

of these, Login is small enough that it's possibly not worth messing with but the other 4 components for sure have enough fields and complexity that it'd be worthwhile; in particular, AddOrder and EditOrder share the child component form that you wrote and would benefit from a shared Formik layout.

### subPR 3: combining ScanOrder, EditOrder and PrintOrder

- pros: unifies the visual layout of these 3 components
- cons: divorces EditOrder and AddOrder not only visually but also breaks up their shared form logic *which would heavily benefit from subPR 2*

ScanOrder, in particular, has a complex hedge maze of nested ternaries to handle various requested use cases.  It's complex out of necessity, not out of virtue.

I can appreciate the visual intent with comboing Scan/Edit/Print but I get Separation of Concerns vibes with the code logic.  It's doable, but it's taking an overly complex component and making it even more so, which in turn makes it *harder* to read, maintain, etc.

## TL;DR

- I think the Button Refactoring and Formik are frickin awesome
- I think comboing Edit/Scan/Print components while divorcing Edit/Add components is scary

File: src/components/ordersView.js [hunk 1]
```diff
@@ -1,22 +1,15 @@
...
+
 import styles from "../style/orders.module.css";
 import AuthService from "../service/authService";
-import { useSortableData } from "./sorting/sortHook";
-import { TableHeader } from "./tableHeader";
-import Gauge from "./gauge";
-import { Search } from "./search";
-import { editOrderControl } from "./protectedRoute/permissions";
-import { useHistory, useLocation } from "react-router-dom";
-import UserContext from "./userContext";
```
Comment (mertbagt) [inline src/components/ordersView.js:19]: do not replace

File: src/components/ordersView.js [hunk 2]
```diff
@@ -1,22 +1,15 @@
-import React, {
-  useState,
-  useEffect,
-  useReducer,
-  useRef,
-  useContext,
```
Comment (mertbagt) [inline src/components/ordersView.js:6]: do not replace ln 6

File: src/components/ordersView.js [hunk 3]
```diff
@@ -196,68 +161,14 @@ const OrdersView = () => {
...
-              </div>
+              <div></div>
             </>
           ))}
     </div>
   );
 
   const closePopUpBox = () => {
     setPopUpBox("none");
-    setUserDisplay();
```
Comment (mertbagt) [inline src/components/ordersView.js:260]: do not replace

File: src/components/selectField.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+//https://gist.github.com/hubgit/e394e9be07d95cd5e774989178139ae8?permalink_comment_id=3366379#gistcomment-3366379
```
Comment (mertbagt) [inline src/components/selectField.js:1]: does this file actually get used anywhere?

I'm trying to figure out the difference between this file:
[src/components/selectField.js](https://github.com/codefordc/us-congress-pizza-flag-tracker-frontend/pull/144/files#diff-f55f82a2a4c9bc2fc1f52d8143f811a6945c33446ca8e43092922b2567ce49d2)

and this one:
[src/components/scanOrder/SelectField.js](https://github.com/codefordc/us-congress-pizza-flag-tracker-frontend/pull/144/files#diff-b1a645be2b714f691a83cec92d7f70071286880d21aae47d76a7d8a4e5609c97)
